Source/Dafny/Verifier/Translator.cs

```diff
@@ -5735,6 +5738,20 @@ private class WFOptions {
             foreach (var ss in TrSplitExpr(precond, etran, true, out splitHappened)) {
               if (ss.IsChecked) {
                 var tok = new NestedToken(expr.tok, ss.E.tok);
+                if (options.VariablesWithCompilableTypeAssumptions.Any() && errorMessage == null) {
+                  foreach (var entry in options.VariablesWithCompilableTypeAssumptions) {
+                    if (e.AllSubExpressions.Any(subExpression =>
+                          subExpression is IdentifierExpr v &&
+                          v.Var.CompileName == entry.Key
+                        )) {
+                      errorMessage = $"possible violation of function precondition." +
```

Comment:
Do I understand correctly that this error message serves as a warning to make other error messages more understandable, because we've changed the set comprehension base type in the verification landscape to help with compilation, so there might be errors that we wouldn't get otherwise?

Reply:
Yes you got it entirely !